api/v1alpha1/stack_type.go

```diff
@@ -4,122 +4,125 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	// the operator is deployed(typically tinkerbell)
-	ImagePullSecret []string `json:"imagePullSecret"`
+	ImagePullSecrets []string `json:"imagePullSecrets,omitempty"`
 
 	// Services contains all Tinkerbell Stack services.
-	Services Services `json:"services"`
+	Services Services `json:"services,omitempty"`
 }
 
 // Services contains all Tinkerbell Stack services.
 type Services struct {
 	// Boots contains all the information and spec about boots.
-	Boots Boots `json:"boots"`
+	Boots Boots `json:"boots,omitempty"`
 
 	// Hegel contains all the information and spec about boots.
-	Hegel Hegel `json:"hegel"`
+	Hegel Hegel `json:"hegel,omitempty"`
 
 	// Rufio contains all the information and spec about rufio.
-	Rufio Rufio `json:"rufio"`
+	Rufio Rufio `json:"rufio,omitempty"`
 
 	// TinkServer contains all the information and spec about tink server.
-	TinkServer TinkServer `json:"tinkServer"`
+	TinkServer TinkServer `json:"tinkServer,omitempty"`
 
 	// TinkController contains all the information and spec about tink controller.
-	TinkController Image `json:"tinkController"`
+	TinkController TinkController `json:"tinkController,omitempty"`
 }
 
 // Boots specifies the deployment details of Tinkerbell service, Boots.
 type Boots struct {
 	// Image specifies the details of a tinkerbell services images
-	Image Image `json:"image"`
+	Image Image `json:"image,omitempty"`
 
 	// DHCPAddressListener set the ip and port to listen on for DHCP.
-	DHCPAddressListener string `json:"dhcpAddress"`
+	DHCPAddressListener string `json:"dhcpAddressListener,omitempty"`
 
 	// TrustedProxies comma separated allowed CIDRs subnets to be used as trusted proxies
-	TrustedProxies string `json:"trustedProxies"`
+	TrustedProxies []string `json:"trustedProxies,omitempty"`
 
 	// HTTPAddress is the address to listen on for the serving iPXE binaries and files via HTTP.
-	HTTPAddress string `json:"httpAddress"`
+	HTTPAddress string `json:"httpAddress,omitempty"`
 
 	// OSIEURL override the URL where OSIE/Hook images are located
-	OSIEURL string `json:"osieURL"`
+	OSIEURL string `json:"osieURL,omitempty"`
 
 	// PublicIP is the IP that netboot clients and/or DHCP relay's will use to reach Boots
-	PublicIP string `json:"publicIP"`
+	PublicIP string `json:"publicIP,omitempty"`
 
 	// PublicSyslogFQDN is the IP that syslog clients will use to send messages
-	PublicSyslogFQDN string `json:"publicSyslogFQDN"`
+	PublicSyslogFQDN string `json:"publicSyslogFQDN,omitempty"`
 
 	// SyslogAddress is the IP and port that syslog clients will use to send messages
-	SyslogAddress string `json:"syslogAddress"`
+	SyslogAddress string `json:"syslogAddress,omitempty"`
 
 	// TinkerbellGRPCAuthority IP and port to listen on for syslog messages.
-	TinkerbellGRPCAuthority string `json:"tinkerbellGRPCAuthority"`
+	TinkerbellGRPCAuthority string `json:"tinkerbellGRPCAuthority,omitempty"`
```

Comment:
The name are copied over from their original naming in their repos: https://github.com/tinkerbell/boots/blob/d135982a3767b050eea20e04615c76b0ef09e0b6/ipxe/script/ipxe.go#L28

Reply:
Recently changed `https://github.com/tinkerbell/boots/blob/d135982a3767b050eea20e04615c76b0ef09e0b6/cmd/boots/flag.go#L103`.

Reply:
Maybe in the checks of the PRs that are opened in the other used repos in the operator, we add a checkbox for updating args in the operator if they changed in their original repos. I guess this would help us having consistency. Will add it in our next community meeting agenda.